sw/legacy/common/CMakeLists.txt

```diff
@@ -4,3 +4,6 @@
 
 add_library(common OBJECT sonata_system.c usbdev.c uart.c timer.c rv_plic.c gpio.c i2c.c pwm.c spi.c crt0.S)
 target_include_directories(common INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}")
+
+
+add_library(syscall OBJECT ../../../vendor/cheriot_debug_module/tb/prog/syscalls.c)
```

Comment:
I wonder if we should copy this file to the `sw/legacy/common`.

Reply:
Are we likely to make changes to this? If so then maybe it's useful to copy, but if we can leave it unchanged we can keep it in the vendor directory.